### ISSUE TITLE
"Awesome Inventory" Achtung + awesome inventory + RT issue

### ISSUE THREAD

**Issue by 136068448** [cseelhoff/RimThreaded#318]
**Describe the bug**
I draft my pawn and order she stand at the spot where have a door, then there have a error occur about UI,i read it and it said the Achtung mod cause it but when i test this bug with Achtung and RT,it not reproduct like i think, then i go back to the save which occur this bug and draft pawn gu to the door, it occur again.i try to disable the mod can change door or add UI all,but still can't solve this question

**To Reproduce**
1. load the mod order
2. build a door
3. draft a pawn and select pawn,then right click the door

**Error Log**
https://gist.github.com/136068448/c3ab8345bd042b56901715b4adad7e1f

**Mod List**
- brrainz.harmony
  - Ludeon.RimWorld
  - Ludeon.RimWorld.Royalty
  - UnlimitedHugs.HugsLib
  - Nals.FacialAnimation
  - erdelf.HumanoidAlienRaces
  - RWZH.ChinesePack.HumanoidAlienRaces
  - OskarPotocki.VanillaFactionsExpanded.Core
  - velc.HatsDisplaySelection
  - brrainz.achtung
  - RWZH.ChinesePack.achtung
  - brrainz.cameraplus
  - RWZH.ChinesePack.cameraplus
  - void.charactereditor
  - UnlimitedHugs.MapReroll
  - UnlimitedHugs.AllowTool
  - fluffy.colonymanager
  - Vanya.Tools.BlightedAlert
  - razor2.3.AnotherRimworldMod.AutoCutBlight
  - Oblitus.AnimalsLogic
  - GHSGKD.AnimalsLogicZH
  - falconne.BWM
  - Mlie.FalloutTraits
  - sarg.smartspeed
  - fluffy.animaltab
  - fluffy.blueprints
  - fluffy.pharmacist
  - fluffy.relationstab
  - fluffy.worktab
  - fluffy.stuffedfloors
  - fluffy.medicaltab
  - leafzxg.MasterOfCooking
  - leafzxg.MasterOfCrafting
  - automatic.gunplay
  - velc.MineItAll
  - tikubonn.DontBlockDoor
  - Ogliss.Mythos.DiseasesOverhauled
  - DiseasesOverhauled.Chinese
  - Troopersmith1.DeathRattle
  - GHSGKD.DeathRattleZH
  - Bar0th.NoRandomRelations
  - Dubwise.DubsMintMenus
  - dubwise.dubsmintminimap
  - WiseClock.AC.DaysMatter
  - fluffy.areaunlocker
  - Mlie.LessArbitrarySurgery
  - Mlie.LessStupidRomanceAttempt
  - smashphil.loadinginprogress
  - brrainz.justignoremepassing
  - tammybee.predatorhuntalert
  - smashphil.carrycapacityfixed
  - Mlie.PickUpAndHaul
  - RWZH.ChinesePack.PrepareLanding
  - neitsa.PrepareLanding
  - Jaxe.Bubbles
  - JaxeZH.Bubbles
  - ChinesePackage.ConfigurableMaps
  - configurablemaps.kv.rw
  - jelly.harvestsync
  - reiquard.researchinfo
  - RWZH.ChinesePack.OgreStack
  - Ogre.OgreStack
  - samo.RareTraits
  - neceros.metaldontburn
  - fed1sPlay.PawnTargetFix
  - com.yayo.raceQuestPawn
  - Ogliss.RescueeJoinsPlus
  - GG.RescueeJoinsPlus
  - Ogliss.WanderJoinsPlus
  - GG.WanderJoinsPlus
  - Vanya.Tools.IncidentPersonStat
  - shavius.yayocombat3.zh
  - com.yayo.combat3
  - crlsniper.realisticrooms
  - GHSGKD.realisticroomsZH
  - Mlie.RealDining
  - ONegative.ReworkedTemperatureExtremeEvents
  - dhultgren.smarterconstruction
  - zlj.staticQualityPlus.124
  - tradingspot.kv.rw
  - RWZH.ChinesePack.tradingspot
  - RWZH.ChinesePack.UnlimitedHugs
  - Linkolas.Stabilize
  - VanillaExpanded.VanillaTraitsExpanded
  - RWZH.ChinesePack.VanillaTraitsExpanded
  - weilbyte.snapout
  - RWZH.ChinesePack.snapout
  - JM.betterTraders
  - com.yayo.BetterJumpPack
  - PeteTimesSix.SimpleSidearms
  - RWZH.ChinesePack.RBSEHC
  - Rah.RBSEHC
  - 400.ImpassableHChestUdeepHWater
  - spoonshortage.ADogSaidAnimalProsthetics
  - Jaxe.RimHUD
  - shouer.chinesepack.RimHUD
  - Mlie.ExpandedIncidents
  - RWZH.ChinesePack.ExpandedIncidents
  - Mlie.MoEvents
  - RWZH.ChinesePack.MoEvents
  - Ameiro.Animehairs
  - Gloomy.GloomyHairMk2
  - GWhair.noface
  - rimsenal.hair
  - sakama.RoppoiHair
  - Orion.XevaRimhair
  - ChJees.Androids
  - RWZH.ChinesePack.Androids
  - RunneLatki.RabbieRaceMod
  - RWZH.ChinesePack.RabbieRaceMod
  - RunneLatki.RabbieVanillafacepatch
  - Solaris.RatkinRaceMod
  - RWZH.ChinesePack.RatkinRaceMod
  - Solaris.RatkinVanillaFace
  - Daemon976.FacialAnimationplus
  - Nals.DynamicPortraits
  - shavius.alienrbse
  - sulusdacor.james76543.Meltup.AdvancedPowerPlus
  - RWZH.ChinesePack.AdvancedPowerPlus
  - Mlie.RunAndHide
  - zh.WallLight
  - Murmur.WallLight
  - Murmur.SubsurfaceConduit
  - hatena.VoiceroidAsAnimal
  - hatti.qualitybuilder
  - RWZH.ChinesePack.qualitybuilder
  - Mlie.SDBridges
  - neceros.plantgrowthsync
  - wit.expandedroofing
  - sarg.alphabiomes
  - QA.QQIJQ
  - dracoix.doormat.r12a
  - factioncontrol.kv.rw
  - Solaris.FurnitureBase
  - RWZH.ChinesePack.FurnitureBase
  - Haplo.Miscellaneous.Endgame
  - rimfridge.kv.rw
  - Albion.SparklingWorlds.Full
  - RWZH.ChinesePack.SparklingWorldsFull
  - GT.Sam.TilledSoil
  - automatic.traderships
  - RWZH.ChinesePack.traderships
  - nephlite.orbitaltradecolumn
  - RWZH.ChinesePack.orbitaltradecolumn
  - VanillaExpanded.VAEAS
  - VanillaExpanded.VAEAU
  - VanillaExpanded.VAEBF
  - VanillaExpanded.VAECD
  - VanillaExpanded.VAED
  - VanillaExpanded.VAEED
  - VanillaExpanded.VAEIS
  - VanillaExpanded.VAETF
  - VanillaExpanded.VAETR
  - VanillaExpanded.VAETS
  - VanillaExpanded.VAET
  - RWZH.ChinesePack.VAE
  - VanillaExpanded.VAPPE
  - VanillaExpanded.VARME
  - VanillaExpanded.VWE
  - VanillaExpanded.VWEQ
  - VanillaExpanded.VWEG
  - RWZH.ChinesePack.VWAAE
  - VanillaExpanded.VCEF
  - RWZH.ChinesePack.VCEF
  - OskarPotocki.VanillaExpanded.RoyaltyPatches
  - VanillaExpanded.VFECore
  - VanillaExpanded.VFEArt
  - VanillaExpanded.VFEFarming
  - VanillaExpanded.VFEMedical
  - VanillaExpanded.VFEProduction
  - VanillaExpanded.VFESecurity
  - VanillaExpanded.VFESpacer
  - RWZH.ChinesePack.VFE
  - akairo.ShibaWanko
  - dismarzero.VGP.VGPVegetableGarden
  - argon.vanillavgarden
  - dismarzero.VGP.VGPGardenFabrics
  - argon.vanillavgardenfab
  - dismarzero.VGP.VGPThreeSistersPlants
  - dismarzero.VGP.VGPMoreVeggies
  - dismarzero.VGP.VGPGardenResources
  - argon.vanillavgardenresources
  - dismarzero.VGP.VGPGardenMedicine
  - dismarzero.VGP.VGPGardenTools
  - shavius.vpe.zh
  - VanillaExpanded.VPlantsE
  - shavius.vbe.zh
  - VanillaExpanded.VBrewE
  - shavius.vce.zh
  - VanillaExpanded.VCookE
  - VanillaExpanded.VCookEStews
  - VanillaExpanded.VCookESushi
  - shavius.vbookse.zh
  - VanillaExpanded.VBooksE
  - OskarPotocki.VanillaFactionsExpanded.MedievalModule
  - RWZH.ChinesePack.MedievalModule
  - OskarPotocki.VanillaFactionsExpanded.SettlersModule
  - RWZH.ChinesePack.SettlersModule
  - shavius.vfeinsectoids.zh
  - OskarPotocki.VFE.Insectoid
  - OskarPotocki.VFE.Vikings
  - RWZH.ChinesePack.Vikings
  - VSE.PerryPersistent
  - RWZH.ChinesePack.PerryPersistent
  - VanillaExpanded.VTEXE
  - VFME.CaravanPacks
  - RWZH.ChinesePack.VTEXE
  - Mlie.RoadsOfTheRim
  - RWZH.ChinesePack.RoadsOfTheRim
  - RWZH.ChinesePack.SlaveOutfits
  - Usgiyi.SlaveOutfits
  - Vanya.Polarisbloc.CoreLab
  - Vanya.Polarisbloc.SecurityForce
  - Mlie.MealRadius
  - Harkon.GunNut.Chapter1
  - Harkon.GunNut.Chapter2
  - Harkon.GunNut.Community
  - RWZH.ChinesePack.GunNut
  - TMC.Armory.AssaultRifle.1
  - TMC.Armory.CommunityPack
  - TMC.Armory.Shotguns.1
  - TMC.Armory.Sidearms.1
  - TMC.Armory.SMGs.1
  - TMC.Armory.RF.1
  - TMC.Armory.SW.1
  - TMC.AAM
  - TMC.Armory.SMGs.1.zhcn
  - TMC.Armory.Sidearms.1.zhcn
  - TMC.Armory.AssaultRifle.1.zhcn
  - TMC.Armory.Shotguns.1.chcn
  - TMC.AAM.zhcn
  - rimsenal.core
  - rimsenal.EVP
  - rimsenal.federation
  - rimsenal.feral
  - rimsenal.storyteller
  - rimsenal.security
  - RWZH.ChinesePack.rimsenal
  - JPT.SoftWarmBeds
  - RWZH.ChinesePack.JPT
  - NotooShabby.AwesomeInventory
  - AwesomeZH.Inventory
  - LingLuo.PrepareCarefully
  - avilmask.CommonSense
  - co.uk.epicguru.whatsthatmod
  - fluffy.modmanager
  - RWZH.ChinesePack.Fluffy
  - notfood.SeedsPlease
  - shadow.spaiomegapatch
  - majorhoff.rimthreaded

**Comment by Niilo007** [comment]
caused by: awesome inventory + Achtung + RT

**Comment by kertiek** [comment]
in the player.log from the reported issue i see this bug

```
Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.ChoicesAtFor_Patch2(UnityEngine.Vector3,Verse.Pawn)
  at AchtungMod.MultiActions.AddColonist (AchtungMod.Colonist colonist) [0x0001f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions.<.ctor>b__3_0 (AchtungMod.Colonist colonist) [0x00000] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Tools.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions..ctor (System.Collections.Generic.IEnumerable`1[T] colonists, UnityEngine.Vector3 clickPos) [0x00018] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.MouseDown (UnityEngine.Vector3 pos) [0x00076] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.HandleEvents () [0x0002f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Selector_HandleMapClicks_Patch.Prefix () [0x00005] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at (wrapper dynamic-method) RimWorld.Selector.RimWorld.Selector.HandleMapClicks_Patch1(RimWorld.Selector)
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimWorld.UIRoot_Play.UIRootOnGUI () [0x000d3] in <d72310b4d8f64d25aee502792b58549f>:0 
  at (wrapper dynamic-method) Verse.Root.Verse.Root.OnGUI_Patch1(Verse.Root)
```
 

... on my customlist that is also using awesome inventory + achtung i was able to reproduce that specific error:

```
Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.ChoicesAtFor_Patch6(UnityEngine.Vector3,Verse.Pawn)
  at AchtungMod.MultiActions.AddColonist (AchtungMod.Colonist colonist) [0x0001f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions.<.ctor>b__3_0 (AchtungMod.Colonist colonist) [0x00000] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Tools.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions..ctor (System.Collections.Generic.IEnumerable`1[T] colonists, UnityEngine.Vector3 clickPos) [0x00018] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.MouseDown (UnityEngine.Vector3 pos) [0x00076] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.HandleEvents () [0x0002f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Selector_HandleMapClicks_Patch.Prefix () [0x00005] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at (wrapper dynamic-method) RimWorld.Selector.RimWorld.Selector.HandleMapClicks_Patch3(RimWorld.Selector)
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <d72310b4d8f64d25aee502792b58549f>:0 
  at (wrapper dynamic-method) RimWorld.UIRoot_Play.RimWorld.UIRoot_Play.UIRootOnGUI_Patch1(RimWorld.UIRoot_Play)
  at (wrapper dynamic-method) Verse.Root.Verse.Root.OnGUI_Patch1(Verse.Root)
Verse.Log:Error(String, Boolean)
Verse.Root:Verse.Root.OnGUI_Patch1(Root)
```

**Comment by kertiek** [comment]
vanilla + achtung + rt, no problems

vnailla + achtung + awesome inventory + rt, selecting multiple panws, right click on a door throws:

```
Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.ChoicesAtFor_Patch2(UnityEngine.Vector3,Verse.Pawn)
  at AchtungMod.MultiActions.AddColonist (AchtungMod.Colonist colonist) [0x0001f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions.<.ctor>b__3_0 (AchtungMod.Colonist colonist) [0x00000] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Tools.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) [0x00014] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.MultiActions..ctor (System.Collections.Generic.IEnumerable`1[T] colonists, UnityEngine.Vector3 clickPos) [0x00018] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.MouseDown (UnityEngine.Vector3 pos) [0x00076] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Controller.HandleEvents () [0x0002f] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at AchtungMod.Selector_HandleMapClicks_Patch.Prefix () [0x00005] in <831e973d9245449bbdebd2f42c6b55fa>:0 
  at (wrapper dynamic-method) RimWorld.Selector.RimWorld.Selector.HandleMapClicks_Patch1(RimWorld.Selector)
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <d72310b4d8f64d25aee502792b58549f>:0 
  at RimWorld.UIRoot_Play.UIRootOnGUI () [0x000d3] in <d72310b4d8f64d25aee502792b58549f>:0 
  at Verse.Root.OnGUI () [0x0003d] in <d72310b4d8f64d25aee502792b58549f>:0 
Verse.Log:Error(String, Boolean)
Verse.Root:OnGUI()

```

**Comment by pastorismylord** [comment]
PooperRulz mentions this issue has been solved on 2.3.6
